gene/etl/ncbi.py

```diff
@@ -0,0 +1,254 @@
+        else:
+            logger.error(f"Entrez gene history download failed with status "
+                         f"code: {response.status_code}")
+            raise DownloadException("Entrez gene history download failed")
+
+    def _files_downloaded(self, data_dir: Path) -> bool:
+        """Check whether needed source files exist.
+
+        :param Path data_dir: source data directory
+        :return: true if all needed files exist, false otherwise
+        """
+        files = data_dir.iterdir()
+
+        info_downloaded: bool = False
+        history_downloaded: bool = False
+
+        for f in files:
+            if f.name.startswith('ncbi_info'):
+                info_downloaded = True
+            elif f.name.startswith('ncbi_history'):
+                history_downloaded = True
+        return info_downloaded and history_downloaded
+
+    def _extract_data(self):
+        """Gather data from local files or download from source.
+        - Data is expected to be in <PROJECT ROOT>/data/ncbi.
+        - For now, data files should all be from the same source data version.
+        """
+        local_data_dir = PROJECT_ROOT / 'data' / 'ncbi'
+        local_data_dir.mkdir(exist_ok=True, parents=True)
+        if not self._files_downloaded(local_data_dir):
+            self._download_data(local_data_dir)
+        local_files = [f for f in local_data_dir.iterdir()
+                       if f.name.startswith('ncbi')]
+        local_files.sort(key=lambda f: f.name.split('_')[-1], reverse=True)
+        self._info_src = [f for f in local_files
+                          if f.name.startswith('ncbi_info')][0]
+        self._history_src = [f for f in local_files
+                             if f.name.startswith('ncbi_history')][0]
+        self._version = self._info_src.stem.split('_')[-1]
+
+    def _transform_data(self):
+        """Modify data and pass to loading functions."""
+        self._add_meta()
+        # get symbol history
+        history_file = open(self._history_src, 'r')
+        history = csv.reader(history_file, delimiter='\t')
+        next(history)
+        prev_symbols = {}
+        for row in history:
+            if row[0] == '9606' and row[1] != '-':
+                gene_id = row[1]
+                if gene_id in prev_symbols.keys():
+                    prev_symbols[gene_id].append(row[3])
+                else:
+                    prev_symbols[gene_id] = [row[3]]
+        history_file.close()
+
+        # open info file, skip headers
+        info_file = open(self._info_src, 'r')
+        info = csv.reader(info_file, delimiter='\t')
+        next(info)
+
+        with self._database.genes.batch_writer() as batch:
+            for row in info:
+                is_valid_row = True
+                params = {
+                    'concept_id': f"ncbigene:{row[1]}",
+                }
+                # get symbol
+                if row[2] != '-':
+                    params['symbol'] = row[2]
```

Comment:
It appears that all genes require a symbol
```suggestion
                params['symbol'] = row[2]
```